inc/utility/class-handlestransfers.php

```diff
@@ -0,0 +1,347 @@
+	 *
+	 * @param $ids
+	 * @return array
+	 */
+	public function getExportableItems( $ids ) {
+		$items = [];
+		$fields = $this->getTransferableFields();
+
+		foreach ( $ids as $id ) {
+			$term = get_term( $id, self::TAXONOMY );
+			$term_meta = get_term_meta( $term->term_id );
+
+			$item = [
+				'name' => $term->name,
+				'slug' => $term->slug,
+			];
+
+			foreach ( $fields as $field ) {
+				$value = $term_meta[ $field ] ?? [];
+
+				$item[ $field ] = $value[0] ?? '';
+			}
+
+			$items[] = $item;
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Download the content as a JSON file.
+	 *
+	 * @param string $content
+	 * @return void
+	 */
+	public function downloadJson( $content ) {
+		$filename = self::TAXONOMY . '-list-' . time() . '.json';
+
+		$file = create_tmp_file();
+
+		put_contents( $file, $content );
+
+		force_download( $file, false, $filename );
+
+		if ( ! defined( 'WP_TESTS_MULTISITE' ) ) {
+			exit;
+		}
+	}
+
+	/**
+	 * Handles file upload before importing contributors.
+	 *
+	 * @return array|false
+	 */
+	public function handleUpload() {
+		if ( empty( $_FILES['import_file']['name'] ) ) {
+			return false;
+		}
+
+		if ( $_FILES['import_file']['type'] !== 'application/json' ) {
+			$_SESSION['pb_errors'][] = __( 'Sorry, this file type is not permitted for security reasons.' );
+
+			return false;
+		}
+
+		$upload = wp_handle_upload( $_FILES['import_file'], [
+			'test_type' => false,
+			'action' => self::TAXONOMY . '-import',
+		] );
+
+		if ( ! empty( $upload['error'] ) ) {
+			$_SESSION['pb_errors'][] = $upload['error'];
+
+			return false;
+		}
+
+		return $upload;
+	}
+
+	/**
+	 * Returns the list of contributors to import
+	 *
+	 * @param array $upload
+	 * @return array
+	 */
+	public function getImportableItems( $upload ) {
+		$items = [];
+		$invalid_rows = [];
+
+		foreach ( json_decode( file_get_contents( $upload['file'] ) ) as $key => $item ) {
+			$item = (array) $item;
+
+			if ( ! isset( $item['name'], $item['slug'] ) ) {
+				$invalid_rows[] = $key + 1;
+
+				continue;
+			}
+
+			$items[] = $item;
+		}
+
+		if ( ! empty( $invalid_rows ) ) {
+			$message = __( 'One or more rows were not imported because they have missing information. Please, check rows %s', 'pressbooks' );
+
+			$_SESSION['pb_errors'][] = sprintf( $message, implode( ',', $invalid_rows ) );
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Imports the list of contributors.
+	 *
+	 * @param array $items
+	 */
+	public function handleImport( $items ) {
+		$importable_fields = $this->getTransferableFields();
+		$changed = false;
+
+		foreach ( $items as $item ) {
+			$term = get_term_by( 'slug', $item['slug'], self::TAXONOMY );
+
+			if ( ! $term ) {
+				$results = wp_insert_term( sanitize_text_field( $item['name'] ), self::TAXONOMY, [
+					'slug' => sanitize_text_field( $item['slug'] ),
+				] );
+			} else {
+				$results = wp_update_term( $term->term_id, self::TAXONOMY, [
+					'name' => sanitize_text_field( $item['name'] ),
```

Comment:
### Not a blocker, but a question

Dalcin, for this validation, and https://github.com/pressbooks/pressbooks/pull/2356/files#diff-9d961ff1ac28242af01296e4dd0a4f7c45f1130737cf9a11f4636add55cf6b8aR278 and https://github.com/pressbooks/pressbooks/pull/2356/files#diff-9d961ff1ac28242af01296e4dd0a4f7c45f1130737cf9a11f4636add55cf6b8aR278 and https://github.com/pressbooks/pressbooks/pull/2356/files#diff-3e98df1fc7a6e8edfe8f3516f827ab9eba589a08edc9a7ab97981fca0b80e57eR394

Would make sense to use a validator (could be a method maybe) that uses: https://github.com/pressbooks/pressbooks/blob/dev/inc/class-contributors.php#L289
Something like iterate on that array and use the unify validation methods we declared there.

Reply:
I think we could have a method that handles input sanitization for sure. Only thinking how to approach this here https://github.com/pressbooks/pressbooks/pull/2356/files#diff-9d961ff1ac28242af01296e4dd0a4f7c45f1130737cf9a11f4636add55cf6b8aR300-R301

Reply:
yes, I would consider it out of the scope. Just wondering if it is easy to add, otherwise the validation is already performed and works.